monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/HttpClientPoolBuilder.java

```diff
@@ -52,15 +53,11 @@ public HttpClientPoolBuilder withDefaultScheduledExecutorService() {
         return this;
     }
 
-    public HttpClientPoolBuilder withSingleHostHttpClientBuilder(final SingleHostHttpClientBuilder singleHostHttpClientBuilder) {
+    public HttpClientPoolBuilder withSingleHostHttpClientBuilder(final Function<InetAddress, SingleHostHttpClientBuilder> singleHostHttpClientBuilder) {
```

Comment:
I am not sure that the name of this method reflects now what it does.

Reply:
it provides the `SingleHostHttpClientBuilder` used to create the client so I think it's still valid. What name would you propose?

Reply:
I would propose not to use function at all (https://github.com/nhenneaux/resilient-httpclient/pull/20#discussion_r366368599),